CMakeLists.txt

```diff
@@ -102,13 +111,12 @@ target_link_libraries(lib_kmeans PRIVATE
    common_flags 
    )
```

Comment:
Why do I need this? Aren't I specifying these directives elsewhere?

Reply:
You have to specify how the CMake consumer should consume `lib_kmeans` (even within the same project). And no, you do not have it specified anywhere. It just accidentally works because everything is built in the same `CMAKE_Fortran_MODULE_DIRECTORY` which is implicitly included. This does not work when other projects try to use your target.

Reply:
hmm, this is new to my limited cmake installation knowledge. I've now added it, but it's not intuitive (for me)